src/Datadog.Trace.ClrProfiler.Managed/Integrations/WebRequestIntegration.cs

```diff
@@ -0,0 +1,117 @@
+        public static object GetResponse(object request)
+        {
+            var webRequest = (WebRequest)request;
+
+            if (!IsTracingEnabled(webRequest))
+            {
+                return webRequest.GetResponse();
+            }
+
+            using (var scope = CreateScope(webRequest, nameof(GetResponse)))
+            {
+                try
+                {
+                    // add distributed tracing headers
+                    webRequest.Headers.Inject(scope.Span.Context);
+
+                    return webRequest.GetResponse();
+                }
+                catch (Exception ex)
+                {
+                    scope.Span.SetException(ex);
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Instrumentation wrapper for <see cref="WebRequest.GetResponseAsync"/>.
+        /// </summary>
+        /// <param name="request">The <see cref="WebRequest"/> instance to instrument.</param>
+        /// <returns>Returns the value returned by the inner method call.</returns>
+        [InterceptMethod(
+            TargetAssembly = "System.Net",
+            TargetType = "System.Net.WebRequest")]
+        public static object GetResponseAsync(object request)
+        {
+            return GetResponseAsyncInternal((WebRequest)request);
+        }
+
+        private static async Task<WebResponse> GetResponseAsyncInternal(WebRequest request)
+        {
+            if (!IsTracingEnabled(request))
+            {
+                return await request.GetResponseAsync().ConfigureAwait(false);
+            }
+
+            using (var scope = CreateScope(request, nameof(GetResponseAsync)))
+            {
+                try
+                {
+                    // add distributed tracing headers
+                    request.Headers.Inject(scope.Span.Context);
+
+                    return await request.GetResponseAsync()
+                                        .ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    scope.Span.SetException(ex);
+                    throw;
+                }
+            }
+        }
+
+        private static bool IsTracingEnabled(WebRequest request)
+        {
+            // check if tracing is disabled for this request via http header
+            string value = request.Headers[HttpHeaderNames.TracingEnabled];
+            return !string.Equals(value, "false", StringComparison.InvariantCultureIgnoreCase);
+        }
+
+        private static Scope CreateScope(WebRequest request, string methodName)
+        {
+            string httpMethod = request.Method.ToUpperInvariant();
+            string url = request.RequestUri.OriginalString;
+            string resourceName = $"{httpMethod} {url}";
+
+            var tracer = Tracer.Instance;
+            var scope = tracer.StartActive(OperationName, serviceName: tracer.DefaultServiceName);
```

Comment:
It would make sense to use a different service name than the default to isolate this from the callee from the caller. Still something that should probably be standardized accross tracers but the nodejs one for instance seems to have either a configured service name or to derive it from the current service: https://github.com/DataDog/dd-trace-js/blob/master/src/plugins/http/client.js#L166

Reply:
I added a `-http-client` suffix like nodejs.